web/auth/auth_config.go

```diff
@@ -0,0 +1,37 @@
+package auth
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+type AuthConfig struct {
+	providers map[string]*oauth2.Config
+}
+
+// AuthConfig creates a new OAuth 2.0 config for every enabled OAuth 2.0 provider.
+// Currently only works for GitHub.
+func NewAuthConfig(baseURL string) (*AuthConfig, error) {
+	providers := make(map[string]*oauth2.Config)
+
+	// Enable GitHub.
+	clientID := os.Getenv("GITHUB_KEY")
```

Comment:
Not sure this is the place to read the env vars. I think it is best to read the env in main.go and pass in these things.